### PR TITLE
tbb: add optional python library

### PR DESCRIPTION
**Pull request by quantumsteve**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

This adds an option to build TBB's new python library that monkey-patches multiprocessing.Pool.

https://software.intel.com/en-us/blogs/2016/04/04/unleash-parallel-performance-of-python-programs

### CODE REVIEW

File: Formula/tbb.rb [hunk 1]
```diff
@@ -16,7 +16,12 @@ class Tbb < Formula
 
   # requires malloc features first introduced in Lion
   # https://github.com/Homebrew/homebrew/issues/32274
-  depends_on :macos => :lion
+  depends_on macos: :lion
```
Comment (MikeMcQuaid) [inline Formula/tbb.rb]: Just leave this as it was for now, thanks.

File: Formula/tbb.rb [hunk 2]
```diff
@@ -16,7 +16,12 @@ class Tbb < Formula
...
+  depends_on python: :optional
+
+  if build.with? "python"
+    depends_on "swig" => :build
```
Comment (MikeMcQuaid) [inline Formula/tbb.rb]: `depends_on "swig" => :build if build.with? "python"`

File: Formula/tbb.rb [hunk 3]
```diff
@@ -17,6 +17,8 @@ class Tbb < Formula
   # requires malloc features first introduced in Lion
   # https://github.com/Homebrew/homebrew/issues/32274
   depends_on :macos => :lion
+  depends_on python: :optional
```
Comment (DomT4) [inline Formula/tbb.rb]: I'd prefer we keep using the hashrocket until we make this a broader change.

Comment (MikeMcQuaid) [inline Formula/tbb.rb]: @DomT4 Reinstate the Rubocop (that I suspect is suggesting people change this)?

Comment (DomT4) [inline Formula/tbb.rb]: Ah, yes, I suspect so. Will double check that locally.

Comment (MikeMcQuaid) [inline Formula/tbb.rb]: @DomT4 https://github.com/Homebrew/brew/pull/736.

**Comment by quantumsteve** [comment]
After updating I noticed that `--with-python` is now generating a [sandbox error](https://gist.github.com/quantumsteve/bf19364cf35201f78c388f787ecfaa1d). How do you recommend addressing this?

**Comment by tdsmith** [comment]
This is a tbb bug; it shouldn't write files outside of its install prefix. I would probably:
- always build this instead of adding an option
- not provide the `python` target to `make`
- install the python module manually

The latter should look something like:
```ruby
cd "python" do
  system "python", *Language::Python.setup_install_args(prefix)
end
```

File: Formula/tbb.rb [hunk 4]
```diff
@@ -17,6 +17,8 @@ class Tbb < Formula
   # requires malloc features first introduced in Lion
   # https://github.com/Homebrew/homebrew/issues/32274
   depends_on :macos => :lion
+  depends_on :python
```
Comment (DomT4) [inline Formula/tbb.rb]: This should probably be `depends_on :python if MacOS.version <= :snow_leopard`

File: Formula/tbb.rb [hunk 5]
```diff
@@ -32,6 +34,11 @@ def install
     system "make", *args
     lib.install Dir["build/BUILDPREFIX_release/*.dylib"]
     include.install "include/tbb"
+
+    cd "python" do
+      ENV["TBBROOT"]=prefix
```
Comment (DomT4) [inline Formula/tbb.rb]: Nit but please leave spacing in formulae usage of this sort of thing:
```ruby
ENV["TBBROOT"] = prefix
```

**Comment by tdsmith** [comment]
Awesome :tada: Thanks for this!

**Pull request closed without merging**